unit_tests/test_StaticSVD.cpp

```diff
@@ -95,7 +108,397 @@ int main(int argc, char* argv[])
 
     }
 
+    return;
+}
+
+TEST(StaticSVDTest, Test_SLPKTranspose)
+{
+    // Get the rank of this process, and the number of processors.
+    int mpi_init, d_rank, d_num_procs;
+    MPI_Initialized(&mpi_init);
+    if (mpi_init == 0) {
+        MPI_Init(nullptr, nullptr);
+    }
+
+    MPI_Comm_rank(MPI_COMM_WORLD, &d_rank);
+    MPI_Comm_size(MPI_COMM_WORLD, &d_num_procs);
+
+    int num_total_rows = 3;
+    int num_total_cols = 5;
+    int d_num_rows = num_total_rows / d_num_procs;
+    if (num_total_rows % d_num_procs > d_rank) {
+        d_num_rows++;
+    }
+    int *row_offset = new int[d_num_procs + 1];
+    row_offset[d_num_procs] = num_total_rows;
+    row_offset[d_rank] = d_num_rows;
+
+    MPI_Allgather(MPI_IN_PLACE,
```

Comment:
Can this be replaced by the new MPI utility function?

Reply:
Replaced